packages/wonder-blocks-core/src/util/add-style.tsx

```diff
@@ -6,14 +6,32 @@ import {processStyleList} from "./util";
 import type {StyleType} from "./types";
 
 export default function addStyle<
+    // We extend `React.ComponentType<any>` to support `addStyle(Link)` with
+    // react-router's `Link` component.
     T extends React.ComponentType<any> | keyof JSX.IntrinsicElements,
->(Component: T, defaultStyle?: StyleType) {
-    type Props = SpreadType<
-        JSX.LibraryManagedAttributes<T, React.ComponentProps<T>>,
-        {style?: StyleType}
-    >;
-
-    const StyleComponent: React.FC<Props> = (props) => {
+    Props extends {
+        className?: string;
+        style?: StyleType;
+        children?: React.ReactNode;
+    } & Omit<React.ComponentProps<T>, "style">, // Removes the 'style' prop from the original component
+>(
+    Component: T,
+    defaultStyle?: StyleType,
+): React.ForwardRefExoticComponent<
+    React.PropsWithoutRef<Props> &
+        React.RefAttributes<
+            // We need to lookup the HTML/SVG element type based on the tag name, but only
+            // for JSX intrinsics (aka HTML/SVG tags).
+            T extends keyof JSX.IntrinsicElements ? IntrinsicElementsMap[T] : T
```

Comment:
I think there might be a way to lookup what we want from `JSX.IntrinsicElements` instead of having to duplicate a list of instrinsics below in `IntrinsicElementsMap`, but I haven't found it yet and this works despite it being a little hacky.